### PR TITLE
Issue 2071. Correctly handle external refs in array items composed sc…

### PR DESCRIPTION
**Pull request by chimmi**
…hemas

### CODE REVIEW

**Comment by chimmi** [comment]
ISSUE-2071

File: modules/swagger-parser-v3/src/test/java/io/swagger/v3/parser/processors/ExternalRefProcessorTest.java [hunk 1]
```diff
@@ -206,4 +211,19 @@ public void testRelativeRefIncludingUrlRef()
         is("./relative-with-url/relative-with-local.yaml#/relative-same-file")
     );
 	}
+
+    @Test
+    public void testHandleComposedSchemasInArrayItems() {
```
Comment (frantuma) [inline modules/swagger-parser-v3/src/test/java/io/swagger/v3/parser/processors/ExternalRefProcessorTest.java:216]: would also test the case with `properties` alongside `oneOf` e.g. with following spec:

```
components:
  schemas:
    Response:
      type: array
      items:
        oneOf:
          - $ref: '#/components/schemas/ProductRow'
        properties: 
          orderRow:
            $ref: '#/components/schemas/OrderRow'
        discriminator:
          propertyName: type
          mapping:
            'product': '#/components/schemas/ProductRow'
    ProductRow:
      type: object
      additionalProperties: false
      required:
        - type
      properties:
        type:
          $ref: '#/components/schemas/ProductsRowType'
        payload:
          type: string
    ProductsRowType:
      type: string
      enum:
        - product    
    OrderRow:
      type: object
      additionalProperties: false
      required:
        - type
      properties:
        type:
          $ref: '#/components/schemas/OrderRowType'
        payload:
          type: string
    OrderRowType:
      type: string
      enum:
        - order

```

**Review by frantuma — CHANGES_REQUESTED** [review]
Thanks for reporting this and for the PR!

Added a minor change request to support `properties` along side `oneOf\anyOf\allOf`

File: modules/swagger-parser-v3/src/test/java/io/swagger/v3/parser/processors/ExternalRefProcessorTest.java (hunk 1 above)
Comment (chimmi) [reply modules/swagger-parser-v3/src/test/java/io/swagger/v3/parser/processors/ExternalRefProcessorTest.java:216]: Did not realise those were not mutually exclusive. Thank you for noticing. Fixed.

**Review by frantuma — APPROVED** [review]

**Pull request merged**